src/context/AuthSync.jsx

```diff
@@ -0,0 +1,34 @@
+import { useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+import { login, logout } from '../store/authFeature/authSlice.js';
+
+export const AuthSync = () => {
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    const token = localStorage.getItem('token');
+    if (token) {
+      dispatch(login());
+    }
+  }, [dispatch]);
+
+  useEffect(() => {
+    const handleStorageChange = (event) => {
+      if (event.key === 'token') {
+        if (event.newValue) {
+          dispatch(login());
+        } else {
+          dispatch(logout());
+        }
+      }
+    };
+
+    window.addEventListener('storage', handleStorageChange);
```

Comment:
instead of using this for getting the token value you can use the window key directly:

https://www.w3schools.com/jsref/prop_win_localstorage.asp

Also create a function under helpers/auth.js

- Name the function as **isLoggedIn** and use that function to check for the user login status (if the user is logged in or not)